b2/bucket.py

```diff
@@ -393,9 +400,60 @@ def _upload_large_file(
         response = self.api.session.finish_large_file(file_id, part_sha1_array)
         return FileVersionInfoFactory.from_api_response(response)
 
+    def _find_unfinished_file(self, upload_source, file_name, file_info, part_ranges):
+        """
+        Find an unfinished file which may be used to resume a large file upload. The
+        file is found using the filename and comparing the uploaded parts against
+        the local file.
+        """
+        for file in self.list_unfinished_large_files():
+            if file.file_name == file_name and file.file_info == file_info:
+                files_match = True
+                finished_parts = {}
+                for part in self.list_parts(file.file_id):
+                    # Compare part sizes
+                    (offset, part_length) = part_ranges[part.part_number - 1]
+                    if part_length != part.content_length:
+                        files_match = False
+                        break
+
+                    # Compare hash
+                    with upload_source.open() as f:
+                        f.seek(offset)
+                        sha1_sum = hex_sha1_of_stream(f, part_length)
+                    if sha1_sum != part.content_sha1:
+                        files_match = False
+                        break
+
+                    # Save part
+                    finished_parts[part.part_number] = part
+
+                # Skip not matching files or unfinished files with no uploaded parts
+                if files_match == False or len(finished_parts) == 0:
+                    continue
+
+                # Return first matched file
+                return (file, finished_parts)
+        return (None, {})
+
     def _upload_part(
-        self, file_id, part_number, part_range, upload_source, large_file_upload_state
+        self,
+        file_id,
+        part_number,
+        part_range,
+        upload_source,
+        large_file_upload_state,
+        finished_parts={}
```

Comment:
this dictionary is global. Common python bug.
To fix it, use `finished_parts=None` and
```python
if finished_parts is None:
    finished_parts = {}
```